vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/BatchRegistrationDialog.java

```diff
@@ -99,60 +94,83 @@ private class RegisterBatchDialogHandler implements Serializable {
 
     @Serial
     private static final long serialVersionUID = -2692766151162405263L;
-    private final List<ComponentEventListener<SampleRegistrationEvent>> listeners = new ArrayList<>();
-    private final List<ComponentEventListener<UserCancelEvent<SampleRegistrationDialog>>> cancelListeners = new ArrayList<>();
+    private final List<ComponentEventListener<SampleRegistrationEvent>> sampleRegistrationListeners = new ArrayList<>();
+    private final List<ComponentEventListener<BatchRegistrationEvent>> batchRegistrationListeners = new ArrayList<>();
+    private final List<ComponentEventListener<UserCancelEvent<BatchRegistrationDialog>>> cancelListeners = new ArrayList<>();
 
     public RegisterBatchDialogHandler() {
+      setNavigationListeners();
+      setSubmissionListeners();
       resetDialogueUponClosure();
-      setBatchInformationButtonsListeners();
-      setSampleRegistrationSubmission();
-      setTabSelectionListener();
     }
 
-    private void setBatchInformationButtonsListeners() {
+    private void setNavigationListeners() {
       batchInformationLayout.nextButton.addClickListener(
           event -> tabStepper.setSelectedTab(sampleInformationTab));
+      setTabSelectionListener();
+    }
+
+    private void setTabSelectionListener() {
+      tabStepper.addSelectedChangeListener(event -> {
+        if (event.getSelectedTab() == sampleInformationTab
+            && batchInformationLayout.isInputValid()) {
+          sampleSpreadsheetLayout.generateSampleRegistrationSheet(
+              batchInformationLayout.dataTypeSelection.getValue());
+          sampleSpreadsheetLayout.setBatchName(batchInformationLayout.batchNameField.getValue());
+        } else {
+          tabStepper.setSelectedTab(batchInformationTab);
```

Comment:
I am not sure what is happening here :/

Reply:
I extracted the samplespreadsheet generation from this method to make it clearer. Basically what's happening is: 
1.) Did the user click on the sample tab in the tabsheet? 
2.) If so did the user provide the correct batch information(datatype and label) 
3A) If that is the case --> Generate spreadsheet and switch to layout with the spreadsheet
3B) If that is not the case --> Keep user on the current tab and don't generate spreadsheet